File: src/main/java/net/lomeli/lomlib/util/BlockUtil.java
package net.lomeli.lomlib.util;

import net.minecraft.block.Block;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.util.math.*;
import net.minecraft.util.EnumFacing;
import net.minecraft.world.IBlockAccess;

import net.minecraftforge.fml.common.registry.GameRegistry;

public class BlockUtil {
    public static void registerTile(Class<? extends TileEntity> tile) {
        GameRegistry.registerTileEntity(tile, tile.getName().toLowerCase());
    }

    public static Block getBlock(IBlockAccess world, BlockPos blockPos) {
        return world.getBlockState(blockPos).getBlock();
    }

    public static Block getBlock(IBlockAccess world, int x, int y, int z) {
        return getBlock(world, new BlockPos(x, y, z));
    }

    public static boolean posEqual(BlockPos pos1, BlockPos pos2) {
        return pos1.getX() == pos2.getX() && pos1.getY() == pos2.getY() && pos1.getZ() == pos2.getZ();
    }

    public static float getDistance(BlockPos pos1, BlockPos pos2) {
        return net.minecraft.util.math.MathHelper.sqrt_double(pos1.distanceSq(pos2.getX(), pos2.getY(), pos2.getZ()));
    }

    public static EnumFacing getFaceFromDif(BlockPos base, BlockPos target) {
        int xDif = target.getX() - base.getX();
        int yDif = target.getY() - base.getY();
        int zDif = target.getZ() - base.getZ();
        if (xDif == 1)
            return EnumFacing.EAST;
        if (xDif == -1)
            return EnumFacing.WEST;
        if (zDif == 1)
            return EnumFacing.SOUTH;
        if (zDif == -1)
            return EnumFacing.NORTH;
        if (yDif == 1)
            return EnumFacing.UP;
        if (yDif == -1)
            return EnumFacing.DOWN;
        return EnumFacing.DOWN;
    }
}

File: src/main/kotlin/net/lomeli/lomlib/util/BlockUtil.kt
package net.lomeli.lomlib.util

import net.minecraft.block.Block
import net.minecraft.tileentity.TileEntity
import net.minecraft.util.EnumFacing
import net.minecraft.util.math.BlockPos
import net.minecraft.util.math.MathHelper
import net.minecraft.world.IBlockAccess
import net.minecraftforge.fml.common.registry.GameRegistry

object BlockUtil {
    fun registerTile(tile: Class<out TileEntity>) {
        GameRegistry.registerTileEntity(tile, tile.name.toLowerCase())
    }

    fun getBlock(world: IBlockAccess, blockPos: BlockPos): Block = world.getBlockState(blockPos).getBlock()

    fun getBlock(world: IBlockAccess, x: Int, y: Int, z: Int): Block = getBlock(world, BlockPos(x, y, z))

    fun posEqual(pos1: BlockPos, pos2: BlockPos): Boolean = pos1.x === pos2.x && pos1.y === pos2.y && pos1.z === pos2.z

    fun getDistance(pos1: BlockPos, pos2: BlockPos): Float = MathHelper.sqrt_double(pos1.distanceSq(pos2.x as Double, pos2.y as Double, pos2.z as Double))

    fun getFaceFromDif(base: BlockPos, target: BlockPos): EnumFacing {
        val xDif = target.x - base.x
        val yDif = target.y - base.y
        val zDif = target.z - base.z
        if (xDif == 1)
            return EnumFacing.EAST
        if (xDif == -1)
            return EnumFacing.WEST
        if (zDif == 1)
            return EnumFacing.SOUTH
        if (zDif == -1)
            return EnumFacing.NORTH
        if (yDif == 1)
            return EnumFacing.UP
        if (yDif == -1)
            return EnumFacing.DOWN
        return EnumFacing.DOWN
    }
}